src/mesh/mesh.hpp

```diff
@@ -168,6 +168,8 @@ class Mesh {
   int GetCurrentLevel() const noexcept { return current_level; }
   std::vector<int> GetNbList() const noexcept { return nblist; }
 
+  void OutputMeshStructure(int dim, bool dump_mesh_structure = true);
```

Comment:
This method should also be made const. As far as i can tell that would mean SetBlockSizeAndBoundaries should also be a const method which calls ComputeMeshGeneratorX. Maybe I'm going down a rabbit hole, but it doesn't make much sense to me that a method that is simply outputting information should not be const.

Reply:
Is it worth fixing this here (also the unique_ptr piece) or cleanup separately?

Reply:
The unique_ptrs are an easy fix, if changing the methods to const leads to a bunch of errors I would just create an issue, if not, might as well fix it now.

Reply:
I addressed the `unique_ptr` piece and `const` method arguments (thanks for insisting, this result in continuously improved code quality!)
With respect to making the function itself `const`, this is less straightforward.
I looked closer and (as you said) `SetBlockSizeAndBoundaries` is called in other places (Mesh generation and AMR) and there it modifies the `loclist` variable, which is a private member variable of `Mesh`.

Reply:
That's what I was worried about, thanks for looking into it.